emily/handler/src/api/models/deposit/requests.rs

```diff
@@ -194,32 +194,44 @@ impl UpdateDepositsRequestBody {
         chainstate: Chainstate,
     ) -> Result<ValidatedUpdateDepositsRequest, error::Error> {
         // Validate all the deposit updates.
-        let mut deposits: Vec<(usize, ValidatedDepositUpdate)> = vec![];
-        let mut failed_txs: Vec<String> = vec![];
+        let mut deposits: Vec<(usize, Result<ValidatedDepositUpdate, ValidationError>)> = vec![];
 
         for (index, update) in self.deposits.into_iter().enumerate() {
             match update
                 .clone()
                 .try_into_validated_deposit_update(chainstate.clone())
             {
-                Ok(validated_update) => deposits.push((index, validated_update)),
-                Err(_) => {
-                    failed_txs.push(format!(
-                        "{}:{}",
-                        update.bitcoin_txid.clone(),
-                        update.bitcoin_tx_output_index
-                    ));
+                Ok(validated_update) => deposits.push((index, Ok(validated_update))),
+                Err(
+                    ref err @ ValidationError::DepositMissingFulfillment(
+                        ref bitcoin_txid,
+                        bitcoin_tx_output_index,
+                    ),
+                ) => {
+                    tracing::warn!(
+                        %bitcoin_txid,
+                        bitcoin_tx_output_index,
+                        "failed to update deposit: request missing fulfillment for completed request."
+                    );
+                    deposits.push((index, Err(err.clone())));
+                }
+                Err(err) => {
+                    tracing::error!(
+                        bitcoin_txid = update.bitcoin_txid,
+                        bitcoin_tx_output_index = update.bitcoin_tx_output_index,
+                        ?err,
```

Comment:
We should call this field `error` to be consistent with the other usage in this PR and elsewhere[^1]. Also, we should use the display implementation instead of the debug implementation, so we should do `%error` instead of `?err`.


[^1]: Well, technically there are some exceptions in `main.rs` in the signer repo, we need to fix that

Reply:
[f36db4f](https://github.com/stacks-sbtc/sbtc/pull/1653/commits/f36db4fdbee7619b946abfa6a11be7ec4ee05c51)